cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -863,3 +863,133 @@ void Refactor::RenameType(mgp_list *args, mgp_graph *memgraph_graph, mgp_result
+
+      mergeRels = config.At(kMergeNodesRelationshipsStrategy).ValueBool();
+    }
+
+    // Get properties strategy from config
+    std::string prop_strategy = std::string(kMergeNodesPropertiesCombine);
+    if (config.KeyExists(kMergeNodesPropertiesStrategy)) {
+      prop_strategy = config.At(kMergeNodesPropertiesStrategy).ValueString();
+    }
+    if (config.KeyExists(kMergeNodesPropertiesStrategyAlternative)) {
+      prop_strategy = config.At(kMergeNodesPropertiesStrategyAlternative).ValueString();
+    }
+
+    const auto mergeNodesPropertiesCombineString = std::string(kMergeNodesPropertiesCombine);
+    const auto mergeNodesPropertiesDiscardString = std::string(kMergeNodesPropertiesDiscard);
+    const auto mergeNodesPropertiesOverrideString = std::string(kMergeNodesPropertiesOverride);
+    const auto mergeNodesPropertiesOverwriteString = std::string(kMergeNodesPropertiesOverwrite);
+
+    // Convert to lowercase for case-insensitive comparison
+    std::transform(prop_strategy.begin(), prop_strategy.end(), prop_strategy.begin(),
+                   [](unsigned char c) { return std::tolower(c); });
+
+    // Validate property strategy
+    if (prop_strategy != mergeNodesPropertiesCombineString && prop_strategy != mergeNodesPropertiesDiscardString &&
+        prop_strategy != mergeNodesPropertiesOverrideString && prop_strategy != mergeNodesPropertiesOverwriteString) {
+      throw mgp::ValueException(std::string(kMergeNodesInvalidPropertyStrategyError).c_str());
+    }
+
+    // Get the first node as the target node
+    auto target_node = nodes[0].ValueNode();
+
+    // Process remaining nodes
+    for (size_t i = 1; i < nodes.Size(); ++i) {
+      auto source_node = nodes[i].ValueNode();
+
+      // Handle properties based on strategy
+      if (prop_strategy == mergeNodesPropertiesCombineString) {
+        // Combine properties from both nodes
+        auto source_props = source_node.Properties();
+        for (const auto &[key, value] : source_props) {
+          if (!source_props.contains(key)) {
+            // nothing to combine
+            continue;
+          }
+
+          auto target_property = target_node.GetProperty(key);
+          if (target_property.IsList()) {
+            auto target_list = target_property.ValueList();
+            target_list.AppendExtend(source_props[key]);
+            target_node.SetProperty(key, mgp::Value(std::move(target_list)));
+          } else if (!target_property.IsNull()) {
+            auto combined_properties = mgp::List();
+            combined_properties.AppendExtend(target_property);
+            combined_properties.AppendExtend(source_props[key]);
+            target_node.SetProperty(key, mgp::Value(std::move(combined_properties)));
+          } else {
+            target_node.SetProperty(key, source_props[key]);
+          }
+        }
+      } else if (prop_strategy == mergeNodesPropertiesDiscardString) {
+        // Keep only target node properties
+        // No action needed
```

Comment:
No need to specify empty block, don't use else if then, write the comment about this case